File: Python/Data Structures & Algorithms Module - Python/Week4/ArrayStackApp.py
"""
ArrayStackApp.py
Billy Cussen
12/11/2020
"""

from ArrayStack import ArrayStack

myArrayStackApp = ArrayStack()

print("Empty: "+str(myArrayStackApp.isEmpty()))
print("Size: "+str(myArrayStackApp.mySize()))
try:
    print("Peek: "+str(myArrayStackApp.peek()))
except Exception as e:
    print(e)
try:
    print("Pop: "+str(myArrayStackApp.pop()))
except Exception as e:
    print(e)
try:
    print(myArrayStackApp.toString())
except Exception as e:
    print(e)

myArrayStackApp.push(2)
myArrayStackApp.push(4)
myArrayStackApp.push(6)
myArrayStackApp.push(8)
myArrayStackApp.push(10)

print("Empty: "+str(myArrayStackApp.isEmpty()))
print("Size: "+str(myArrayStackApp.mySize()))
try:
    print("Peek: "+str(myArrayStackApp.peek()))
except Exception as e:
    print(e)
try:
    print("Pop: "+str(myArrayStackApp.pop()))
except Exception as e:
    print(e)
try:
    print(myArrayStackApp.toString())
except Exception as e:
    print(e)

while myArrayStackApp.isEmpty() == False:
    myArrayStackApp.pop()

print("Empty: "+str(myArrayStackApp.isEmpty()))
print("Size: "+str(myArrayStackApp.mySize()))
try:
    print("Peek: "+str(myArrayStackApp.peek()))
except Exception as e:
    print(e)
try:
    print("Pop: "+str(myArrayStackApp.pop()))
except Exception as e:
    print(e)
try:
    print(myArrayStackApp.toString())
except Exception as e:
    print(e)

File: Java/Data Structures & Algorithms Module - Java/Week8/Revision/Week4/ArrayStackApp.java
/*
ArrayStackApp.java
Billy Cussen
12/11/2020
*/
public class ArrayStackApp{

    public static void main(String [] args){
        StackInterface<Integer> myArrayStack = new ArrayStack<Integer>(2);

        System.out.println("\n***ARRAY STACK***\n");
        System.out.println("Is Empty: "+myArrayStack.isEmpty());
        try{
            System.out.println("Size: "+myArrayStack.size());
        }catch(Exception e){
            System.out.println(e.getMessage());
        }
        try{
            System.out.println("Peek: "+myArrayStack.peek());
            }catch(Exception e){
                System.out.println(e.getMessage());
        }
        try{
            System.out.println("Pop: "+myArrayStack.pop());
            }catch(Exception e){
                System.out.println(e.getMessage());
        }
        System.out.println(myArrayStack.toString()+"\n");

        myArrayStack.push(2);
        myArrayStack.push(4);
        myArrayStack.push(6);
        myArrayStack.push(8);
        myArrayStack.push(10);
        System.out.println("Is Empty: "+myArrayStack.isEmpty());
        try{
            System.out.println("Size: "+myArrayStack.size());
        }catch(Exception e){
            System.out.println(e.getMessage());
        }
        try{
            System.out.println("Peek: "+myArrayStack.peek());
            }catch(Exception e){
                System.out.println(e.getMessage());
        }
        try{
            System.out.println("Pop: "+myArrayStack.pop());
            }catch(Exception e){
                System.out.println(e.getMessage());
        }
        System.out.println(myArrayStack.toString());

        while(!myArrayStack.isEmpty()){
            myArrayStack.pop();
        }
        System.out.println("Is Empty: "+myArrayStack.isEmpty());
        try{
            System.out.println("Size: "+myArrayStack.size());
        }catch(Exception e){
            System.out.println(e.getMessage());
        }
        try{
            System.out.println("Peek: "+myArrayStack.peek());
            }catch(Exception e){
                System.out.println(e.getMessage());
        }
        try{
            System.out.println("Pop: "+myArrayStack.pop());
            }catch(Exception e){
                System.out.println(e.getMessage());
        }
        System.out.println(myArrayStack.toString());
    }
}